experiment/template/eVOLVER.py

```diff
@@ -0,0 +1,629 @@
+                                                    od_cal[0,x])-1)) /
+                                                    od_cal[3,x]))
+                    if not np.isfinite(od_data[x]):
+                        od_data[x] = 'NaN'
+                        logger.debug('OD from vial %d: %s' % (x, od_data[x]))
+                    else:
+                        logger.debug('OD from vial %d: %.3f' % (x, od_data[x]))
+                else:
+                    logger.error('OD calibration not formatted correctly')
+                    od_data[x] = 'NaN'
+            except ValueError:
+                print("OD Read Error")
+                logger.error('OD read error for vial %d, setting to NaN' % x)
+                od_data[x] = 'NaN'
+            try:
+                temp_data[x] = (float(temp_data[x]) *
+                                temp_cal[0][x]) + temp_cal[1][x]
+                logger.debug('temperature from vial %d: %.3f' % (x, temp_data[x]))
+            except ValueError:
+                print("Temp Read Error")
+                logger.error('temperature read error for vial %d, setting to NaN'
+                            % x)
+                temp_data[x]  = 'NaN'
+            try:
+                set_temp_data[x] = (float(set_temp_data[x]) *
+                                    temp_cal[0][x]) + temp_cal[1][x]
+                logger.debug('set_temperature from vial %d: %.3f' % (x,
+                                                                set_temp_data[x]))
+            except ValueError:
+                print("Set Temp Read Error")
+                logger.error('set temperature read error for vial %d, setting to NaN'
+                            % x)
+                set_temp_data[x]  = 'NaN'
+
+        temps = np.array(temps)
+        # update temperatures only if difference with expected
+        # value is above 0.2 degrees celsius
+        delta_t = np.abs(set_temp_data - temps).max()
+        if delta_t > 0.2:
+            logger.info('updating temperatures (max. deltaT is %.2f)' %
+                        delta_t)
+            raw_temperatures = [str(int((temps[x] - temp_cal[1][x]) /
+                                        temp_cal[0][x]))
+                                for x in vials]
+            self.update_temperature(raw_temperatures)
+        else:
+            # config from server agrees with local config
+            # report if actual temperature doesn't match
+            delta_t = np.abs(temps - temp_data).max()
+            if delta_t > 0.2:
+                logger.info('actual temperature doesn\'t match configuration '
+                            '(yet? max deltaT is %.2f)' % delta_t)
+                logger.debug('temperature config: %s' % temps)
+                logger.debug('actual temperatures: %s' % temp_data)
+
+        # add a new field in the data dictionary
+        data['transformed'] = {}
+        data['transformed']['od_90'] = od_data
```

Comment:
We should eventually make this generalized (od_135, other experimental params). This is good for now though.